codegen/src/main/kotlin/apifi/codegen/CodeGenerator.kt

```diff
@@ -9,9 +9,9 @@ class CodeGenerator {
         val modelFiles: List<FileSpec> = if (spec.models.isNotEmpty()) listOf(ModelFileBuilder.build(spec.models, basePackageName)) else emptyList()
         val modelMapping = modelFiles.flatMap { it.members.mapNotNull { m -> (m as TypeSpec).name }.map { name -> name to "${it.packageName}.$name" } }.toMap()
         val apiGroups = spec.paths.groupBy { it.operations?.firstOrNull()?.tags?.firstOrNull() }.filter { it.key != null }
-
-        val apiBuilder = ApiBuilder()
-        val apiClassFiles = apiGroups.map { apiBuilder.build(it.key!!, it.value, basePackageName, modelMapping) }
+        val securityProvider = SecurityProvider(spec.securityDefinitions, spec.securityRequirements)
+        val apiBuilder = ApiBuilder(ApiMethodBuilder(modelMapping, securityProvider), basePackageName, securityProvider)
```

Comment:
@bhimsenp Can we do some change here? I see we are passing`securityProvider` to `ApiBuilder` because `ControllerInterfaceBuilder` needs it. We can remove `securityProvider` and replace it with `ControllerInterfaceBuilder` in `ApiBuilder`'s constructor. so `ApiBuilder(apiMethodBuilder, controllerInterfaceBuilder, basePackageName)`